### ISSUE TITLE
Create an official docker container

### ISSUE THREAD

**Issue by aignas** [mapbox/tippecanoe#593]
Hello all,

I see that there is a dockerfile in the repository and it can be used in order to build a containerized version of the tool and by default it is running the tests.

It would be really nice to have an automated build setup on dockerhub, which would always have the up to date version of tippecanoe. It would allow one to just pull the container and use a known to work version.

Are there any plans on doing something like this?

Thanks.

**Comment by springmeyer** [comment]
> I see that there is a dockerfile in the repository and it can be used in order to build a containerized version of the tool and by default it is running the tests.

Yes, it is used to ensure tippecanoe works as expected when run within docker.

> It would be really nice to have an automated build setup on dockerhub, which would always have the up to date version of tippecanoe. It would allow one to just pull the container and use a known to work version.

We've been considering this, so I appreciate your thoughts. What we are weighing is that the time spent maintaining a docker image on dockerhub takes away from the limited time that we have for maintaining and developing the core of tippecanoe. So, it would be helpful to know more details about what exactly you see valuable about us providing an official docker container. 

Is it mostly so you don't have to guess/research/known which version of tippecanoe is the best to use and most stable?

Or is it mostly to have an automatic way to follow the latest development without needing to increment a version yourself?

Or is it mostly to save you the time of building the image yourself?

I recognizes its probably a combination of all those things - and other considerations perhaps - but what I'm getting at in asking for more detail is to consider whether we can address some of those needs in a way that does not add to our maintenance burden.

**Comment by aignas** [comment]
>
> Is it mostly so you don't have to guess/research/known which version of
> tippecanoe is the best to use and most stable?
>
> Or is it mostly to have an automatic way to follow the latest development
> without needing to increment a version yourself?
>
> Or is it mostly to save you the time of building the image yourself?
>
> I recognizes its probably a combination of all those things - and other
> considerations perhaps - but what I'm getting at in asking for more detail
> is to consider whether we can address some of those needs in a way that
> does not add to our maintenance burden.
>
I would say that it is mostly a combination of the second and third point.
To setup the CI for building the container is easiest when doing it from
your repository. At the moment there are 2 Dockerfiles one for CentOS and
one for Ubuntu. Setting up Travis to push the docker images would not be
hard to do and it would for once make it easier for people to get started
with `tippecanoe`.

As you say this time maintaining the Dockerfile might take away some time
from the core development, however I see this packaging as something that
could be done with the help of the community as well. I would be willing to
invest some time to get this thing rolling. I could create a PR with a
simple Dockerfile that builds everything in docker and then strips all of
the intermediate compilation files and source and has the `tippecanoe` as
main entry point. However, that might mean that the test `Dockerfile` may
be renamed to something else.

Would you be interested in this?

**Comment by josiekre** [comment]
I would find this useful. I was looking for tippecanoe on Docker Hub because I did not want to clone this repo.

In the end, I need a machine with python3.7 on it too since that's where I process all my files from. So I may need to create a custom one anyway.
